core/arch/arm/kernel/virtualization.c

```diff
@@ -0,0 +1,32 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <compiler.h>
+#include <platform_config.h>
+#include <kernel/panic.h>
+#include <kernel/virtualization.h>
+#include <sm/optee_smc.h>
+
+static uint16_t current_client_id = 0;
+
+uint32_t client_created(uint16_t client_id)
```

Comment:
It's guests we're dealing with here, I'd prefer `virt_guest_created()` and so on for the other functions

Reply:
Ah. Okay. I used "client" because SMCCC uses this term for virtual OSes and also it is used in `struct thread_ctx`.

I'll rework my PR and also I'll renamed field `hyp_clnt_id` in `struct thread_ctx` for uniformity.

Reply:
I don't see a problem with letting the parameter have the name `client_id` as that's the name of the field. It's just that the function isn't keeping track of clients (using that terminology here would be very confusing).